pop-api/examples/dao/src/lib.rs

```diff
@@ -0,0 +1,455 @@
+	pub struct Transaction {
+		// The recipient of the proposal
+		pub beneficiary: AccountId,
+		// Amount of tokens to be awarded to the beneficiary
+		pub amount: Balance,
+	}
+
+	/// Structure of a DAO (Decentralized Autonomous Organization)
+	/// that uses Psp22 to manage the Dao treasury and funds projects
+	/// selected by the members through governance
+	#[derive(Default)]
+	#[ink(storage)]
+	pub struct Dao {
+		// Funding proposals
+		proposals: Mapping<u32, Proposal>,
+
+		// Mapping of AccountId to Member structs, representing DAO membership.
+		members: Mapping<AccountId, Member>,
+
+		// Mapping tracking the last time each account voted.
+		last_votes: Mapping<AccountId, BlockNumber>,
+
+		// Duration of the voting period
+		voting_period: BlockNumber,
+
+		// Identifier of the Psp22 token associated with this DAO
+		token_id: TokenId,
+
+		// Proposals created in the history of the Dao
+		proposals_created: u32,
+	}
+
+	/// Defines an event that is emitted
+	/// every time a member voted.
+	#[derive(Debug)]
+	#[ink(event)]
+	pub struct Voted {
+		pub who: Option<AccountId>,
+		pub when: Option<BlockNumber>,
+	}
+
+	impl Dao {
+		/// Instantiate a new Dao contract and create the associated token
+		///
+		/// # Parameters:
+		/// - `token_id` - The identifier of the token to be created
+		/// - `voting_period` - Amount of blocks during which members can cast their votes
+		/// - `min_balance` - The minimum balance required for accounts holding this token.
+		// The `min_balance` ensures accounts hold a minimum amount of tokens, preventing tiny,
+		// inactive balances from bloating the blockchain state and slowing down the network.
+		#[ink(constructor, payable)]
+		pub fn new(
+			token_id: TokenId,
+			voting_period: BlockNumber,
+			min_balance: Balance,
+		) -> Result<Self, Psp22Error> {
+			let instance = Self {
+				proposals: Mapping::default(),
+				members: Mapping::default(),
+				last_votes: Mapping::default(),
+				voting_period,
+				token_id,
+				proposals_created: 0,
+			};
+			let contract_id = instance.env().account_id();
+			api::create(token_id, contract_id, min_balance).map_err(Psp22Error::from)?;
+			instance.env().emit_event(Created {
+				id: token_id,
+				creator: contract_id,
+				admin: contract_id,
+			});
+
+			Ok(instance)
+		}
+
+		/// Allows members to create new spending proposals
+		///
+		/// # Parameters
+		/// - `beneficiary` - The account that will receive the payment
+		/// if the proposal is accepted.
+		/// - `amount` - Amount requested for this proposal
+		/// - `description` - Description of the proposal
+		#[ink(message)]
+		pub fn create_proposal(
+			&mut self,
+			beneficiary: AccountId,
+			amount: Balance,
+			mut description: Vec<u8>,
+		) -> Result<(), Error> {
+			let caller = self.env().caller();
+			let contract = self.env().account_id();
+			self.proposals_created = self.proposals_created.saturating_add(1);
```

Comment:
I suggest moving this line to the bottom of the function, above the event.

Reply:
I use it to change the proposal below.